gcc/rust/typecheck/rust-hir-type-check-pattern.cc

```diff
@@ -43,32 +43,97 @@ TypeCheckPattern::Resolve (HIR::Pattern *pattern, TyTy::BaseType *parent)
 void
 TypeCheckPattern::visit (HIR::PathInExpression &pattern)
 {
-  infered = TypeCheckExpr::Resolve (&pattern);
-
-  /*
-   * We are compiling a PathInExpression, which can't be a Struct or Tuple
-   * pattern. We should check that the declaration we are referencing IS NOT a
-   * struct pattern or tuple with values.
-   */
+  // Pattern must be enum variants, sturcts, constants, or associated constansts
```

Comment:
Is this original comment incorrect? I thought PathInExpression could only reference a unit variant (not tuple variant or struct variant).

Reply:
Or would you get a Hir::PathInExpression for unit structs, aka
struct Unit;

Reply:
Oh the original message is okay but it lacks constants and associated constants

Reply:
I just split this comment to one here and one in line 100

Reply:
I guess my only problem is that instead of 
" Pattern must be enum variants, sturcts, constants, or associated constansts"
It should be
" Pattern must be enum variants, unit structs, constants, or associated constants"
right?

Your new comment is more concise which is nice